bazel/external/quiche.BUILD

```diff
@@ -4247,38 +4249,44 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "quic_core_session_lib",
-    srcs = [
-        "quiche/quic/core/legacy_quic_stream_id_manager.cc",
-        "quiche/quic/core/quic_control_frame_manager.cc",
-        "quiche/quic/core/quic_crypto_handshaker.cc",
-        "quiche/quic/core/quic_crypto_stream.cc",
-        "quiche/quic/core/quic_datagram_queue.cc",
-        "quiche/quic/core/quic_flow_controller.cc",
-        "quiche/quic/core/quic_session.cc",
-        "quiche/quic/core/quic_stream.cc",
-        "quiche/quic/core/quic_stream_id_manager.cc",
-        "quiche/quic/core/quic_stream_sequencer.cc",
-        "quiche/quic/core/tls_handshaker.cc",
-        "quiche/quic/core/uber_quic_stream_id_manager.cc",
-    ],
-    hdrs = [
-        "quiche/quic/core/handshaker_delegate_interface.h",
-        "quiche/quic/core/legacy_quic_stream_id_manager.h",
-        "quiche/quic/core/quic_control_frame_manager.h",
-        "quiche/quic/core/quic_crypto_client_stream.h",  # required by tls_client_handshaker.h
-        "quiche/quic/core/quic_crypto_handshaker.h",
-        "quiche/quic/core/quic_crypto_stream.h",
-        "quiche/quic/core/quic_datagram_queue.h",
-        "quiche/quic/core/quic_flow_controller.h",
-        "quiche/quic/core/quic_session.h",
-        "quiche/quic/core/quic_stream.h",
-        "quiche/quic/core/quic_stream_id_manager.h",
-        "quiche/quic/core/quic_stream_sequencer.h",
-        "quiche/quic/core/stream_delegate_interface.h",
-        "quiche/quic/core/tls_client_handshaker.h",  # required by tls_handshaker.cc
-        "quiche/quic/core/tls_handshaker.h",
-        "quiche/quic/core/uber_quic_stream_id_manager.h",
-    ],
+    srcs = envoy_select_enable_http3(
```

Comment:
Is this change needed? It looks like none of the existing QUIC targets in bazel/external/quiche.BUILD are wrapped in envoy_select_enable_http3(). Instead, I think we tend to use envoy_select_enable_http3() up in envoy code.  Should we perhaps we consistent with the other QUIC targets in this file?

Reply:
This will be used to prevent future regression given we can no longer rely on FIPS build to detect unintended leakage. And it seems that only one quic targets using envoy_select_enable_http3() is sufficient instead of all of them.

Reply:
I like the idea of adding some logic to make sure we don't accidentally leak QUIC code into the no-QUIC build.

If we're going to go this select route, it seems like it would be a good idea to do this for all the various QUIC targets. Wrapping just "quic_core_session_lib" means that it'll break if someone tries to use the code in quic_core_session_lib, or a higher-level target, but it wouldn't break if someone tried to use a *lower* level target (like quic_framer.h, for example). 

What if we did something like:

```
def envoy_quic_cc_library(                                                                                                                                                                          
        name,                                                                                                                                                                                  
        srcs = [],                                                                                                                                                                             
        hdrs = [],                                                                                                                                                                             
        deps = []):                                                                                                                                                                        
    native.cc_library(                                                                                                                                                                         
        name = name,                                                                                                                                                                           
        srcs = envoy_select_enable_http3(srcs),                                                                                                                                                                           
        hdrs = envoy_select_enable_http3(hdrs),                                                                                                                                                                           
        repository = "@envoy",
        copts = quiche_copts,
        tags = ["nofips"],
        visibility = ["//visibility:public"]
        deps = envoy_select_enable_http3(deps),             
    )                                                                                                                                                                                          
```
Then for all the various QUIC targets in this file we could just do:
```
envoy_quic_cc_library(...)
```
And it would automatically do the select for us?

Reply:
Adding envoy_quic_cc_library is a good idea. But batch writer also depends on some lower level QUIC libraries (i.e. packet writer interface) and it is supposed to be a non-QUIC extension.  So I guess we have to leave some QUIC libraries unchanged.

Reply:
Curious. We have some QUIC code that's present in Envoy even when we're not speaking QUIC? Sounds like we should consider moving that code out of QUIC and into QUICHE proper. That being said, in the meantime leaving those libraries unchanged SGTM

Reply:
QuicBatchWriter was implemented with QUIC interfaces, but it can be used for any UDP listeners.